integration_test/third_party_apps_test.go

```diff
@@ -510,14 +478,40 @@ const (
 	SAPHANAApp      = "saphana"
 )
 
+// incompatibleOperatingSystem looks at the supported_operating_systems field
+// of metadata.yaml for this app and returns a nonempty skip reason if it
+// thinks this app doesn't support the given platform.
+// supported_operating_systems should only contain "linux", "windows", or
+// "linux_and_windows".
+func incompatibleOperatingSystem(t *testing.T, testCase test) string {
+	supported := testCase.metadata.SupportedOperatingSystems
+	if supported == "linux_and_windows" {
+		return "" // This app supports both Linux and Windows.
+	}
+	if supported == "windows" {
+		if !gce.IsWindows(testCase.platform) {
+			return fmt.Sprintf("Skipping test for platform %v because app %v only supports Windows.", testCase.platform, testCase.app)
```

Comment:
#719 introduced `gce.PlatformKind()` that could make this code simpler:
```
func incompatibleOperatingSystem(t *testing.T, testCase test) string {
	switch supported := testCase.metadata.SupportedOperatingSystems; supported {
	case "linux_and_windows":
		return "" // This app supports both Linux and Windows.
	case "windows", "linux":
		if supported != gce.PlatformKind(testCase.platform) {
			return fmt.Sprintf("Skipping test for platform %v because app %v only supports %v.", testCase.platform, testCase.app, supported)
		}
		return "" // We are testing on a supported platform for this app.
	default:
		t.Fatalf("Unrecognized value %q for supported_operating_systems.", supported)
		return ""
	}
}
```
or even (if we choose to ignore the unrecognized platform case and rely on the validation):
```
func incompatibleOperatingSystem(t *testing.T, testCase test) string {
	supported := testCase.metadata.SupportedOperatingSystems
	if ! strings.Contains(supported, gce.PlatformKind(testCase.platform)) {
		return fmt.Sprintf("Skipping test for platform %v because app %v only supports %v.", testCase.platform, testCase.app, supported)
	}
	return "" // We are testing on a supported platform for this app.
}
```

Reply:
done